### PR TITLE
Fix top-level exports corner cases.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Review by tanishiking — APPROVED** [review]
I left a few questions, but overall it looks good to me :+1:
Having some comments (something like commit message on https://github.com/tanishiking/scala-wasm/pull/128/commits/208202ab62a3f7ab81ab1af205dabe79c4483e76) at `transformTopLevelExport` would be helpful

**Comment by sjrd** [comment]
I added comments as suggested.

File: wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/Emitter.scala [hunk 1]
```diff
@@ -182,25 +182,31 @@ final class Emitter(config: Emitter.Config) {
...
+            instrs += wa.CALL(genFunctionName.makeExportedDef)
           }
-        case TopLevelFieldExportDef(_, _, _) =>
-          // Nothing to do
-          ()
+        case TopLevelFieldExportDef(_, _, fieldIdent) =>
+          /* Usually redundant, but necessary if the static field is never
+           * explicitly set and keeps its default (zero) value instead. In that
+           * case this initial call is required to publish that zero value (as
+           * opposed to the default `undefined` value of the JS `let`).
```
Comment (tanishiking) [inline wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/Emitter.scala:203]: 👍

**Review by tanishiking — APPROVED** [review]
LGTM, thanks for the explanations!

**Pull request merged**